### DISCUSSION TITLE
Remove lodaconfirmmodal in bootstrap

### DISCUSSION THREAD

**Discussion by chrish112** [PrivateBin/PrivateBin#1381] [Q&A] [answered]
### Describe the problem/question

I wish to remove the following code from the boostrap.php file on PrivateBin 1.7.3. The problem is that by removing the popup in the center of the webpage it instead becomes a web browser popup. If going about the removal of the popup after clicking the privatebin link is correct then how do I remove the web browser popup?

		<div id="loadconfirmmodal" tabindex="-1" class="modal fade" role="dialog" aria-hidden="true">
			<div class="modal-dialog" role="document">
				<div class="modal-content">
					<div class="modal-header">
						<button type="button" class="close" data-dismiss="modal" aria-label="<?php echo I18n::_('Close') ?>"><span aria-hidden="true">&times;</span></button>
						<h4 class="modal-title"><?php echo I18n::_('This secret message can only be displayed once. Would you like to see it now?') ?></h4>
					</div>
					<div class="modal-body text-center">
						<button id="loadconfirm-open-now" type="button" class="btn btn-success" data-dismiss="modal"><span class="glyphicon glyphicon-download"></span> <?php echo I18n::_('Yes, see it') ?></button>
					</div>
				</div>
			</div>
		</div>

### Did you use the FAQ section?

- [X] Yes, I have read [the FAQ](https://github.com/PrivateBin/PrivateBin/wiki/FAQ) and I found no solution/answer there.

### What you did?

**You put the "!--" before div**	<div id="loadconfirmmodal" tabindex="-1" class="modal fade" role="dialog" aria-hidden="true">
			<div class="modal-dialog" role="document">
				<div class="modal-content">
					<div class="modal-header">
						<button type="button" class="close" data-dismiss="modal" aria-label="<?php echo I18n::_('Close') ?>"><span aria-hidden="true">&times;</span></button>
						<h4 class="modal-title"><?php echo I18n::_('This secret message can only be displayed once. Would you like to see it now?') ?></h4>
					</div>
					<div class="modal-body text-center">
						<button id="loadconfirm-open-now" type="button" class="btn btn-success" data-dismiss="modal"><span class="glyphicon glyphicon-download"></span> <?php echo I18n::_('Yes, see it') ?></button>
					</div>
				</div>
			</div>
**Put "--" at the end of div**		</div>

### What happens

The pop-up goes away from the center of the screen but instead turns into a web browser pop-up 
![Screenshot 2024-07-25 100118](https://github.com/user-attachments/assets/e32e6c42-1d3b-4380-8325-0ee75d7499b9)

### What should happen

The button entire button should be removed.

### Server OS

Ubuntu 22 Server

### Webserver

Apache2

### PrivateBin version

v1.7.3

### Issue reproducibility

No, I cannot reproduce it on https://privatebin.net.

**Comment by elrido** [comment, marked as answer]
As you discovered, for templates that don't have modals (a bootstrap CSS specific widget type), we fall back to a classic pop-up dialogues, which will work in all cases. This behaviour is "baked in" (as of 1.7.0), not configurable and triggered if the [paste URL is of the "burn-after-reading" type](https://github.com/PrivateBin/PrivateBin/wiki/Encryption-format#url-format-of-the-paste).

If you don't want your paste to display that message, you can, on pastes that you created yourself, manually remove that dash (`-`) character after the hash / pound (`#`) character in the URL. When someone opens the URL without that dash, the paste will immediately get loaded and deleted on the server (the pre-1.7.0 behaviour).

If you wanted to revert this behaviour on your own instance, you'd have to comment or remove these lines in the JavaScript code and [generate a fresh SRI-hash for the file](https://github.com/PrivateBin/PrivateBin/wiki/FAQ#how-to-make-privatebin-work-when-i-have-changed-some-javascript-files):

https://github.com/PrivateBin/PrivateBin/blob/41369e28d56bbf971fac02d6821d8e56ed041936/js/privatebin.js#L5450-L5454

Unlike a simple template change, that will require re-doing this effort upon every update.

**Comment by chrish112** [comment]
As always, thank you El Rido.
________________________________
From: El RIDO ***@***.***>
Sent: Thursday, July 25, 2024 11:09:33 AM
To: PrivateBin/PrivateBin ***@***.***>
Cc: chrish112 ***@***.***>; Author ***@***.***>
Subject: [ADV][EXT] Re: [PrivateBin/PrivateBin] Remove lodaconfirmmodal in bootstrap (Discussion #1381)


As you discovered, for templates that don't have modals (a bootstrap CSS specific widget type), we fall back to a classic pop-up dialogues, which will work in all cases. This behaviour is "baked in" (as of 1.7.0), not configurable and triggered if the paste URL is of the "burn-after-reading" type<https://github.com/PrivateBin/PrivateBin/wiki/Encryption-format#url-format-of-the-paste>.

If you don't want your paste to display that message, you can, on pastes that you created yourself, manually remove that dash (-) character after the hash / pound (#) character in the URL. When someone opens the URL without that dash, the paste will immediately get loaded and deleted on the server (the pre-1.7.0 behaviour).

If you wanted to revert this behaviour on your own instance, you'd have to comment or remove these lines in the JavaScript code and generate a fresh SRI-hash for the file<https://github.com/PrivateBin/PrivateBin/wiki/FAQ#how-to-make-privatebin-work-when-i-have-changed-some-javascript-files>:

https://github.com/PrivateBin/PrivateBin/blob/41369e28d56bbf971fac02d6821d8e56ed041936/js/privatebin.js#L5450-L5454

Unlike a simple template change, that will require re-doing this effort upon every update.
